### PR TITLE
Palvelutapahtuma

### PR DESCRIPTION
**Pull request by mikajylha**

### CODE REVIEW

File: input/fsh/examples/encounterPalvelutapahtuma.fsh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+* id = "id-for-palvelutapahtuma"
+* identifier.use = #official
+* identifier.value = "1.2.246.10.6794809.14011.2023.200035"
+* status = #in-progress
+* class.system = "http://terminology.hl7.org/ValueSet/v3-ActEncounterCode"
+* class.code = #IMP
+* class.display = "inpatient encounter"
+* subject.reference = "Patient/id-for-patient"
+* period.start = "2022-02-27T01:03:00+02:00"
+* serviceProvider.reference = "Organization/1.2.246.10.8286189.10.100012"
```
Comment (mrinnetmaki) [inline input/fsh/examples/encounterPalvelutapahtuma.fsh:15]: I would not mark the reference like this. This indicates that the oid is the `id` of the resource. This results in 
```
"[serviceProvider](http://hl7.org/fhir/R4/encounter.html#Encounter.serviceProvider)" : {
    "[reference](http://hl7.org/fhir/R4/references.html#Reference#Reference.reference)" : "Organization/1.2.246.10.8286189.10.100014"
  },
  ```
  
I would expect the `oid` to be an identifier, and therefore would like to see
```
"serviceProvider" : {
    "type" : "ServiceProvider",
    "identifier": "1.2.246.10.8286189.10.100014"
  },```

All this assuming that the purpose of using the oid here is to indicate that we may actually not have a full FHIR resource available for the organization. Or even if we did and would populate the `.reference` property too, we can add the identifier to communicate the oid.

Comment (mrinnetmaki) [reply input/fsh/examples/encounterPalvelutapahtuma.fsh:15]: Just a note, we could add the name of the service provider to the reference too.

File: input/pagecontent/StructureDefinition-fi-base-encounter-intro.md [hunk 2]
```diff
@@ -1,10 +1,50 @@
...
+First thing is to identify the appropriate aggregation level of encounter. If encounter is not
+representing a *palvelutapahtuma*, but is a lower level encounter (some systems call these
+*prosessitapahtuma*), parentOf -should be used to point to upper level encounter (up until
+*palvelutapahtuma* -level is reached).
+
+*Palvelutapahtuma* -encounter is identified by identifier. When encounter has identifier with
+`use=official` it SHOULD be considered as palvelutapahtuma and that identifier should be the
+OID of *palvelutapahtuma*.
+
+Other levels of encounter that are not *palvelutapahtuma*, MUST not contain `use=official` identifier.
```
[the remark this answers is not in the document]
Comment (mrinnetmaki) [reply input/pagecontent/StructureDefinition-fi-base-encounter-intro.md:47]: with a capital NOT

**Review by mrinnetmaki — CHANGES_REQUESTED** [review]
Can we get polished references with identifiers?

Also, one "MUST not" should be "MUST NOT".

File: input/pagecontent/StructureDefinition-fi-base-encounter-intro.md [hunk 3]
```diff
@@ -1,10 +1,50 @@
...
+#### Why does encounter need this information?
+
+Kanta HL7 V3 Medical Records specification requires that both queries and archivals transmit
+*palvelutapahtuma*'s OID identifier on each request. A client application that is integrated to a
+patient administration system (one that masters the data of encounters) often needs to create and
+query Kanta Medical records. Encounter is the best "vessel" we have to transmit this information.
+
+For example a laboratory information system may have it's own Kanta Medical Records capabilities
+and will archive lab results directly to Kanta. It receives encounter id in SMART App Launch
+context. Laboratory system can resolve Palvelutapahtuma`s OID identifier by fetching the encounter
```
[the remark this answers is not in the document]
Comment (mikajylha) [reply input/pagecontent/StructureDefinition-fi-base-encounter-intro.md]: Fixed

File: input/pagecontent/StructureDefinition-fi-base-encounter-intro.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikajylha) [reply input/pagecontent/StructureDefinition-fi-base-encounter-intro.md]: fixed

File: input/fsh/examples/encounterPalvelutapahtuma.fsh (hunk 1 above)
Comment (mikajylha) [reply input/fsh/examples/encounterPalvelutapahtuma.fsh:15]: Uses identifiers now

**Review by mrinnetmaki — APPROVED** [review]
I changed the case of `reference.type`s and changed the first example to use such a format for the reference too. OK?

**Pull request merged**